tests/test_core.py

```diff
@@ -259,5 +259,18 @@ def test_get_subset():
     x = _get_subset(df, C=['control', 'hist'], D=['NO2'], require_all_on='B')
     assert x.empty
 
+    x = _get_subset(df, C=['control', 'hist'], D=['O2', 'NO2'], require_all_on='B')
+    assert x.empty
+
```

Comment:
Do you think it would be worth adding 
```
_get_subset(df, C=['hist'], D=['NO2', 'O2'], require_all_on='B')

expected = [
         {'A': 'IPSL', 'B': 'FOO', 'C': 'hist, 'D': 'NO2'},
         {'A': 'IPSL', 'B': 'FOO', 'C': 'hist', 'D': 'O2'},
     ]
```
just to make sure this works on different columns?

Reply:
Good catch. I added a few more tests including this one.